packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/__tests__/FileSystemGraphObjectStore.test.ts

```diff
@@ -129,6 +129,25 @@ describe('flushEntitiesToDisk', () => {
     const symlinkedData = await fs.readFile(expectedIndexFilePath, 'utf8');
     expect(symlinkedData).toEqual(writtenStepData);
   });
+
+  test('should only flush entities that match the passed in stepId', async () => {
+    const stepIdA = 'a';
+    const { store } = setupFileSystemObjectStore(undefined, stepIdA);
+    const entityTypeA = uuid();
+    const entitiesA = times(3, () => createTestEntity({ _type: entityTypeA }));
+    await store.addEntities(stepIdA, entitiesA);
+
+    const stepIdB = 'b';
+    const entityTypeB = uuid();
+    const entitiesB = times(3, () => createTestEntity({ _type: entityTypeB }));
+    await store.addEntities(stepIdB, entitiesB);
+
+    await store.flushEntitiesToDisk(
+      // eslint-disable-next-line @typescript-eslint/require-await
```

Comment:
How about something like this, so that the test will fail if the callback was never called:

```typescript
const onEntitiesFlushedFn = jest.fn().mockResolvedValue(Promise.resolve());
await store.flushEntitiesToDisk(onEntitiesFlushedFn);

expect(onEntitiesFlushedFn).toHaveBeenCalledTimes(1);
expect(onEntitiesFlushedFn).toHaveBeenCalledWith(entitiesA);
```

Reply:
More explicit. Much like. Will change.